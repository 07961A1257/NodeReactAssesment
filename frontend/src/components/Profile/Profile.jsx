import React, { useEffect, useState } from "react";
// import { useSearchParams } from "react-router-dom";
import axios from "../../apis/axios";
import ErrorComponent from "./../ErrorComponent";

const Profile = () => {
  const [loading, setLoading] = useState(false);
  const [user, setUser] = useState([]);
  const [error, setError] = useState("");
  // const [searchParams] = useSearchParams();

  useEffect(() => {
    // console.log(searchParams.get("id"));
    const fetchProfile = async () => {
      // const id = searchParams.get("id");
      const id = sessionStorage.getItem("USER_ID");
      setLoading(true);
      axios
        .get(`/${id}`)
        .then((res) => {
          if (res.status === 200) {
            setUser(res.data);
            setError("");
          }
        })
        .catch((err) => {
          setUser([]);
          setError(err.data);
          console.log(err);
        })
        .finally(() => {
          setLoading(false);
        });
    };

    fetchProfile();
  }, []);

  return (
    <div className="container mt-5">
      <div className="row justify-content-left">
        <div className="col-12">
          <h2>Profile</h2>
        </div>
      </div>
      {loading && <h4>Loading</h4>}
      {error && <ErrorComponent errorMessage={error} />}
      {user.length === 0 && <h4>No user found</h4>}
      {user.length > 0 && (
        <>
          <div className="row justify-content m-5">
            <div className="col-2">
              <h4>Name:</h4>
            </div>
            <div className="col-2 text-left">{user[0].name}</div>
          </div>
          <div className="row justify-content m-5">
            <div className="col-2">
              <h4>Email:</h4>
            </div>
            <div className="col-2 text-left">{user[0].email}</div>
          </div>
          <div className="row justify-content m-5">
            <div className="col-2">
              <h4>Role:</h4>
            </div>
            <div className="col-2 text-left">{user[0].role}</div>
          </div>
        </>
      )}
    </div>
  );
};

export default Profile;
