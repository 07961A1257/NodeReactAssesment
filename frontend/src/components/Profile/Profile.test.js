import React from "react";
import Profile from "./Profile";
import { render } from "@testing-library/react";

it("should match snapshot", () => {
  const { asFragment } = render(<Profile />);

  expect(asFragment()).toMatchSnapshot();
});
