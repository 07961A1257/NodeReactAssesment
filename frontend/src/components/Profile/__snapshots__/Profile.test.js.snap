// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="row justify-content-left"
    >
      <div
        class="col-12"
      >
        <h3>
          Profile
        </h3>
      </div>
    </div>
    <h4>
      Loading
    </h4>
    <h4>
      No user found
    </h4>
  </div>
</DocumentFragment>
`;
